tdrs-backend/tdpservice/scheduling/tasks.py

```diff
@@ -0,0 +1,112 @@
+"""schedule tasks."""
+from __future__ import absolute_import
+# The tasks
+
+import hashlib
+import os
+
+from celery import shared_task
+from django.conf import settings
+import datetime
+import paramiko
+import logging
+from tdpservice.data_files.models import DataFile, LegacyFileTransfer
+
+logger = logging.getLogger(__name__)
+
+
+def write_key_to_file(private_key):
+    """Paramiko require the key in file object format."""
+    with open('temp_key_file', 'w') as f:
+        f.write(private_key)
+        f.close()
+    return 'temp_key_file'
+
+
+@shared_task
+def upload(data_file_pk,
+           server_address=settings.ACFTITAN_SERVER_ADDRESS,
+           local_key=settings.ACFTITAN_LOCAL_KEY,
+           username=settings.ACFTITAN_USERNAME,
+           port=22
+           ):
+    """
+    Upload to SFTP server.
+
+    This task uploads the file in DataFile object with pk = data_file_pk
+    to sftp server as defined in Settings file
+    """
+    # Upload file
+    data_file = DataFile.objects.get(id=data_file_pk)
+    file_transfer_record = LegacyFileTransfer(
+        data_file=data_file,
+        uploaded_by=data_file.user,
+        file_name=data_file.create_filename(),
+    )
+
+    def create_dir(directory_name, sftp_server):
+        """Code snippet to create directory in SFTP server."""
+        try:
+            sftp_server.chdir(directory_name)  # Test if remote_path exists
+        except IOError:
+            sftp_server.mkdir(directory_name)  # Create remote_path
+            sftp_server.chdir(directory_name)
+
+    try:
+        # Create directory names for ACF titan
+        destination = str(data_file.create_filename())
+        today_date = datetime.datetime.today()
+        upper_directory_name = today_date.strftime('%Y%m%d')
+        lower_directory_name = today_date.strftime(str(data_file.year) + '-' + str(data_file.quarter))
+
+        # Paramiko need local file
+        f = data_file.file.read()
+        with open(destination, 'wb') as f1:
+            f1.write(f)
+            file_transfer_record.file_size = f1.tell()
+            file_transfer_record.file_shasum = hashlib.sha256(f).hexdigest()
+            f1.close()
+
+        # Paramiko SSH connection requires private key as file
+        temp_key_file = write_key_to_file(local_key)
+        os.chmod(temp_key_file, 0o600)
+
+        # Create SFTP/SSH connection
+        transport = paramiko.SSHClient()
+        transport.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        pkey = paramiko.RSAKey.from_private_key_file(temp_key_file)
+        transport.connect(server_address,
+                          pkey=pkey,
+                          username=username,
+                          port=port,
+                          look_for_keys=False,
+                          disabled_algorithms={'pubkeys': ['rsa-sha2-512', 'rsa-sha2-256']})
+        # remove temp key file
```

Comment:
```suggestion

```

Reply:
This is the comment for next step, why do you want to remove it?